bin/main.rb

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require_relative '../lib/game_manager.rb'
+
+class Main
+  def initialize
+    @game = GameManager.new
+
+    @board = @game.init_board(9)
+    @player1 = nil
+    @player2 = nil
+  end
+
+  def welcome
+    puts "\nWelcome to Tic Tac Toe\n\n"
+
+    print 'Player 1 please enter your name: '
+    @player1 = gets.chomp
+
+    print "\nPlayer 2 please enter your name: "
+    @player2 = gets.chomp
+
+    puts "\nPlayer 1: #{@player1}, Token: X"
+    puts "Player 2: #{@player2}, Token: O\n"
+
+    display
+  end
+
+  def ending
+    puts "\n\nThank you for playing Tic Tac Toe."
+  end
+
+  def display
+    puts "\n       #{@board[0]} | #{@board[1]} | #{@board[2]} "
+
+    puts '       ---------- '
+
+    puts "       #{@board[3]} | #{@board[4]} | #{@board[5]} "
+
+    puts '       ---------- '
+
+    puts "       #{@board[6]} | #{@board[7]} | #{@board[8]} "
+  end
+
+  def turn
+    current = @game.current_player
+    if current == 'X'
+      print "\n #{@player1}, choose a spot between 1-9: "
+    else
+      print "\n #{@player2}, choose a spot between 1-9: "
+    end
+    spot = gets.strip
+    spot = @game.board_index(spot)
+    if @game.valid_move?(spot)
```

Comment:
This method might be doing to many things, can you describe it's objective with one single line without using any "and" in between? what does "turn" do?  let's see:

 It checks which player turns it is asking for a prompt **and** displays the board

There you go, the method can be divided in more methods, each method should have one single responsibility.

--------------------------------------------

> Methods, like classes, should have a single responsibility. All of the same reasons
> apply; having just one responsibility makes them easy to change and easy to reuse. All
> the same design techniques work; ask them questions about what they do and try to
> describe their responsibilities in a single sentence. 

> 
>  Methods that have a single responsibility confer the following benefits:
> • **Expose previously hidden qualities** Refactoring a class so that all of its methods
> have a single responsibility has a clarifying effect on the class. Even if you do not
> intend to reorganize the methods into other classes today, having each of them
> serve a single purpose makes the set of things the class does more obvious.
> • **Avoid the need for comments** How many times have you seen a comment that is
> out of date? Because comments are not executable, they are merely a form of
> decaying documentation. If a bit of code inside a method needs a comment,
> extract that bit into a separate method. The new method name serves the same
> purpose as did the old comment.
> • **Encourage reuse** Small methods encourage coding behavior that is healthy for
> your application. Other programmers will reuse the methods instead of duplicating the code. They will follow the pattern you have established and create small,
> reusable methods in turn. This coding style propagates itself.
> • **Are easy to move to another class** When you get more design information and
> decide to make changes, small methods are easy to move. You can rearrange behavior
> without doing a lot of method extraction and refactoring. Small methods lower
> the barriers to improving your design.


_**ref: Practical Object-Oriented Design in Ruby: An Agile Primer by Sandi Metz**_